transport/http/outbound.go

```diff
@@ -227,77 +220,40 @@ func (o *Outbound) CallOneway(ctx context.Context, treq *transport.Request) (tra
 		return nil, intyarpcerrors.AnnotateWithInfo(yarpcerrors.FromError(err), "error waiting for http oneway outbound to start for service: %s", treq.Service)
 	}
 
-	start := time.Now()
-	deadline, _ := ctx.Deadline()
-	ttl := deadline.Sub(start)
-
-	_, err := o.call(ctx, treq, start, ttl)
+	_, err := o.call(ctx, treq)
 	if err != nil {
 		return nil, err
 	}
 
 	return time.Now(), nil
 }
 
-func (o *Outbound) call(ctx context.Context, treq *transport.Request, start time.Time, ttl time.Duration) (*transport.Response, error) {
-	p, onFinish, err := o.getPeerForRequest(ctx, treq)
-	if err != nil {
-		return nil, err
-	}
-
-	resp, err := o.callWithPeer(ctx, treq, start, ttl, p)
-
-	// Call the onFinish method right before returning (with the error from call with peer)
-	onFinish(err)
-	return resp, err
-}
+func (o *Outbound) call(ctx context.Context, treq *transport.Request) (*transport.Response, error) {
+	start := time.Now()
+	deadline, _ := ctx.Deadline()
```

Comment:
can you add a comment about why it's safe to ignore whether the deadline is set here

Reply:
We wrap all outbounds with a [validator outbound](https://github.com/yarpc/yarpc-go/blob/dev/internal/request/validator_outbound.go) which ensures that a context is set on each call.

However, I've added the check here again, since it's possible for users to keep a reference to a created outbound and directly use `Call(...)` (and therefore circumventing the validator outbound).